File: April-18/07-04-18/java/01-Calculate-the-power.java

import java.util.*;


class TestClass {
    static long modularExponentiation(long x,long n,long M)
    {
        long result=1;
        while(n>0)
        {
            if(n % 2 ==1)
                result=(result * x)%M;
            x=(x*x)%M;
            n=n/2;
        }
        return result;
    }
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        long a,b;
        a = input.nextLong();
        b = input.nextLong();
        System.out.println(modularExponentiation(a,b,1000000007));
    }
}

File: April-18/07-04-18/python/01-Calculate-the-power.py
def modularExponentiation(a, n, M):
    result = 1
    while(n > 0):
        if(n % 2 == 1):
            result = (result * a) % M;
        a = (a*a) % M;
        n = n // 2
    return result
    
a , b = map(int,input().split())
print(modularExponentiation(a,b,1000000007))
